agentio/__init__.py

```
@@ -0,0 +1,202 @@
+
+You should have received a copy of the GNU Lesser General Public
+License along with this library; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+Linden Research, Inc., 945 Battery Street, San Francisco, CA  94111  USA
+
+$LicenseInfo:firstyear=2022&license=viewerlgpl$
+Copyright (c) 2022, Linden Research, Inc.
+$/LicenseInfo$
+"""
+
+CONTROLLER_PUMP = 'agentio.controller'
+
+#Globals
+_commandRegistry = {}       #List of callbacks for handling messages.
+_controller = None          #The class handling leapIO (EX puppetry)
+
+_look_at=None
+_camera=None
+_agent_orientation=None
+
+def registerCommand(command, func=None):
+    """ decorator usage: @registerCommand('command')
+                  def do_command(...):  ..."""
+    def _register(fn):
+        global _commandRegistry
+        _commandRegistry[command] = fn
+        return fn
+    if func is not None:
+        # function call usage: register('command', func)
+        return _register(func)
+    return _register
+
+def deregisterCommand(command):
+    """ not really used, but here for completeness.  Removes a command from handled data """
+    global _commandRegistry
+    try:
+        del _commandRegistry[command]
+    except:
+        _controller.log(f"failed deregister command='{command}'")
+
+def __init__():
+    pass
+
+def _handleCommand(message):
+    """  Process message as a dict from the viewer
+    message = { data: { command: 'foo', args: { ... } }, pump: ... }
+    """
+
+    _controller._logger.debug(f"leap has command '{message}'")
+    handled = False
+    try:
+        command_name = message['data']['command']
+        command_args = message['data'].get("args", {})
+        try:
+            operator = _commandRegistry[command_name]
+            if operator:
+                try:
+                    operator(command_args)
+                    handled = True
+                except Exception as e:
```

Is there a more specific error this should be catching?

---

Fixed.

---

Hmm, I don't think this addresses the suggestion. `except Exception as e` will catch all exceptions, which is the same behavior as before. Is there a more specific class or classes of expected exceptions?